### ISSUE TITLE
Type not found: BIGINT

### ISSUE THREAD

**Issue by PaddyAlton** [googleapis/python-bigquery-sqlalchemy#73]
Hello! I've been trying to make use of `pybigquery` so that in future our applications can be more datastore agnostic.

Issue summary
--------------------

I am using `pybigquery` in conjunction with `pandas`, so instead of using `pandas.DataFrame.to_gbq` I have been using `pandas.DataFrame.to_sql` using the `sqlalchemy` engine returned by `create_engine("bigquery://")`. I have `if_exists="append"` set. This engine generally seems to work well for retrieving data, but when the first `to_sql` call gets made (which ought to create a new table in the pre-prepared dataset) I get an error:

`Type not found: BIGINT at ...`

The short version is, an attempt to create a table programmatically tries to create fields with type `BIGINT` instead of `INT64` (this is not supported by BigQuery). It would be good if engines created with `pybigquery` caught and fixed this.

Further details
-------------------

I discovered [a similar problem here](https://stackoverflow.com/questions/60218232/pandas-sqlalchemy-pybigquery-specifying-bigint-instead-of-int64-for-inte), although as far as I can tell the questioner never raised an issue on this repository.

Ultimately, this is handled by a `pandas.io.sql.SQLTable` object that gets used to represent a DataFrame. There is a step in there that maps pandas types to sqlalchemy types. `SQLTable._create_table_setup` returns a `sqlalchemy.Table` and invokes `SQLTable._get_column_names_and_types`, which utilises `SQLTable._sqlalchemy_type`, which hits this block:

```python
        elif col_type == "integer":
            if col.dtype == "int32":
                return Integer   # [MY COMMENT] this is sqlalchemy.types.Integer
            else:
                return BigInteger # [MY COMMENT] this is sqlalchemy.types.BigInteger
```

To cut the story short, the method `sqlalchemy.Table.create` is called on the output. Thus, the following SQL instruction is being passed to BigQuery:

```
CREATE TABLE `<dataset>.<table>` (
    ...
    `some_field` BIGINT,
    ...
)
```

and this exception bubbles up:

```python
google.cloud.bigquery.dbapi.exceptions.DatabaseError: 400 GET https://bigquery.googleapis.com/bigquery/v2/projects/<PROJECT ID>/queries/<QUERY ID>?maxResults=0&location=US&prettyPrint=false: Type not found: BIGINT at [<line number>:<character number>]
```

The stackoverflow question I linked does report a (somewhat laborious) workaround to make the `CREATE TABLE` use `INT64` instead of `BIGINT` by using `sqlalchemy.types.Integer` as the datatype of the column in the pandas DataFrame - but personally I'd probably revert to the `DataFrame.to_gbq` method before resorting to this. Would it be possible to instead have `pybigquery` substitute the types behind the scenes? At a minimum, it would seem useful for the plugin to raise its own exception if a column's type is not one that's supported by BigQuery. Even better if `BigInteger` can be made to represent `INT64` in the context of a `pybigquery` engine.

It may be that other data types are affected, [I found this table of corresponding types that might help diagnose others](https://www.striim.com/docs/en/bigquery-data-type-support-and-correspondence.html).

**Comment by tswast** [comment]
> Even better if BigInteger can be made to represent INT64 in the context of a pybigquery engine.

I think is the ideal solution. I see we have a mapping from BigQuery types to SQLAlchemy types

https://github.com/mxmzdlv/pybigquery/blob/2a41ed7ab1c8fc7e97c11e871520933bf795c87e/pybigquery/sqlalchemy_bigquery.py#L93

But I don't see any logic to go in the other direction yet.

**Comment by tswast** [comment]
I suspect we need to override

https://github.com/sqlalchemy/sqlalchemy/blob/3e49b8d0519aa024842206a2fb664a4ad83796d6/lib/sqlalchemy/sql/compiler.py#L4316 

in our

https://github.com/mxmzdlv/pybigquery/blob/2a41ed7ab1c8fc7e97c11e871520933bf795c87e/pybigquery/sqlalchemy_bigquery.py#L194

Although, if I understand https://github.com/sqlalchemy/sqlalchemy/blob/50417cf758508e970f3fccc99af6a501ceac8eb2/lib/sqlalchemy/dialects/type_migration_guidelines.txt correctly, it's saying we should actually override https://github.com/sqlalchemy/sqlalchemy/blob/50417cf758508e970f3fccc99af6a501ceac8eb2/lib/sqlalchemy/dialects/type_migration_guidelines.txt to return `return self.visit_INTEGER(type_, **kw)`

**Comment by jimfulton** [comment]
@PaddyAlton I believe this is fixed in 0.7.0.  Are you still seeing a problem?
